pkg/distributor/validator.go

```diff
@@ -39,8 +43,7 @@ type validationContext struct {
 	userID string
 }
 
-func (v Validator) getValidationContextFor(userID string) validationContext {
-	now := time.Now()
+func (v Validator) getValidationContextFor(now time.Time, userID string) validationContext {
```

Comment:
Minor quibble here:
Argument order is sometimes a proxy for importance; I'd argue that `userID` is the "main" argument here, which `now` is far less important. If this method were named `getValidationContextForTime` then I'd buy the argument ordering here

Reply:
Yea that make sense. Fixed.